apps/prairielearn/src/question-servers/freeform.ts

```diff
@@ -679,25 +718,35 @@ async function traverseQuestionAndExecuteFunctions(phase, codeCaller, data, cont
         }
       }
     }
-    const newChildren = [];
+
+    // Comment nodes don't have childNodes and can't be serialized.
+    if (!('childNodes' in node)) return null;
+
+    const newChildren: parse5.DefaultTreeAdapterTypes.ChildNode[] = [];
     for (let i = 0; i < (node.childNodes || []).length; i++) {
       const childRes = await visitNode(node.childNodes[i]);
       if (childRes) {
         if (childRes.nodeName === '#document-fragment') {
           newChildren.push(...childRes.childNodes);
-        } else {
+        } else if ('tagName' in childRes) {
```

Comment:
`#text` nodes don't have a tag name but should still be processed.

Reply:
I moved these changes over to #11838.